File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // promt the user for the text
    string text = get_string("Text: ");

    // calaculates the number of letters
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    // calaculates the number of words
    int words = 0;
    for (int j = 0; j < strlen(text); j++)
    {
        if ((text[j] == ' ' && text[j] != 0))
        {
            words++;
        }
    }
    words++;

    // caclculates the number of sentences
    int sentences = 0;
    for (int k = 0; k < strlen(text); k++)
    {
        if (text[k] == '.' || text[k] == '?' || text[k] == '!')
        {
            sentences++;
        }
    }
    // printf("%i \n",letters);
    // printf("%i \n",words);
    // printf("%i \n",sentences);

    // claculate number of letters per 100 words
    int L = (float) letters / (float) words * 100;

    // claculate number of sentences per 100 words
    int S = (float) sentences / (float) words * 100;

    // compute the readability level through Coleman-Liau index.
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // if condition about the dagree of the readability
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

# prompt the user for input
text = get_string("Enter the text:")

# claculate how many sentences
sentences = 0
for c in text:
    if (c == '?' or c == '!' or c == '.'):
        sentences = sentences + 1

# claculate how many words
words = 1
for c in text:
    if (c == ' ' and c != 0):
        words = words + 1

# claculate how many letters
letters = 0
for c in text:
    if (c.isalpha()):
        letters = letters + 1

# average number of letrers per (100) words
L = letters / words * 100

# average number of sentences per (100) words
S = sentences / words * 100

# calculate the index
index = round(0.0588 * L - 0.296 * S - 15.8)


# print the grade
if (index > 1 and index < 16):
    print(f"Grade {index}")

elif (index > 16):
    print("Grade 16+")

else:
    print("Before Grade 1")
